hyperledger/iroha
# Fix address sanitizer issue with FairMstProcessor

### Description of the Change

For reasons unknown address sanitizer reported use after free on iroha shutdown, with MST enabled.
The problem was with unsubscribing from observable of GossipPropagationStrategy, which occured in FairMstProcessor destructor. I was not able to replicate this issue, or determine why it occured. 

My suspicion is that it is a false-positive from a address sanitizer. Changing initialization of observable seem to resolve this issue.

How to verify that it works:
1. Build irohad with address sanitizer
2. Set mst_enable to true in config
3. Run irohad
4. Shutdown irohad (Ctrl-C), address sanitizer should not report any issues.

Please verify this PR using the guide above.



### Benefits

No reports from address sanitizer


### Possible Drawbacks 

It is not clear why this has been occuring.
